### PR TITLE
revamped details screen

### PR DESCRIPTION
**Pull request by AndyScherzinger**
successor of #1127

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky @jancborchardt this is ready for review, please see screenshots (with theming active). What I changed:

* I use the default headline style (see red text) for all dialogs (like Google)
* I display the sharees (no avatars accessible at the moment)
* I put the favorite toggle in the view, so the user can change it by simply hitting the toggle

I am aware that this is not the final stage of the detail screen where Jan (and I) want to get to but it is a first step towards the goal and imho fine to merge/ship since it is an improvement to today's version.

device-2017-07-13-181901
device-2017-07-13-181916
device-2017-07-13-182002
device-2017-07-13-182524

What do you think?

**Comment by tobiasKaminsky** [comment]
Is it fine for you to postpone this to 1.6? Then we can focus for now on finalizing 1.5.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky well, I don't see a reason why to postpone it since auto upload will need some more days/weeks to be done and this is basically complete and just needs review/approval and is fixing some display bugs like displaying file meta data correctly (using table layout instead of Frame) and also fixing some minor theming issues.

Other than that sure, while it would be then ineffective again since we might hit merge conflicts down the road in 2-3 months while now we can simply merge after review...

**Comment by tobiasKaminsky** [comment]
I'll have a look into it next week.

**Comment by jancborchardt** [comment]
Looks good for now. :) We can have another round of design review after my vacation.

**Comment by AndyScherzinger** [comment]
@jancborchardt sounds good, maybe having a sitdown during the hackweek at the conference in August :)

**Comment by tobiasKaminsky** [comment]
> no avatars accessible at the moment

This is possible via this url, back to nc10: https://cloud.cloud.com/avatar/$username/$pixel
It will return either an image or {"data":{"displayname":"tobi"}} if the user set no image.

I opened an issue about this: https://github.com/nextcloud/server/issues/5750

**Comment by AndyScherzinger** [comment]
I'd vote for putting this into a new issue since this would also impact the sharee screens / user search. Or do you think this should be added to this PR right away? (because then it is likely 1.6.x) ;)

**Comment by tobiasKaminsky** [comment]
You are right, we create a new & independent PR for showing avatars. Then this can be implemented for the whole app. :+1: 
(I'll create an issue)

File: src/main/res/values/strings.xml [hunk 1]
```diff
@@ -522,6 +522,9 @@
     <string name="share_known_remote_clarification">%1$s ( at %2$s )</string>
 
     <string name="share_sharee_unavailable">Upgrade the server version to allow sharing between users from within their clients.\nPlease contact your administrator</string>
+    <string name="filedetails_share_link_enable">Share by link enabled</string>
+    <string name="filedetails_share_link_disable">Share by link disabled</string>
```
Comment (tobiasKaminsky) [inline src/main/res/values/strings.xml]: disabled sounds for me like it is disabled/disallowed on server-side.
But it is meant that there is currently no link for this particular file/folder, right?
So maybe "Not shared by link"?

In the next iteration we can enhance this that the link is then copied to clipboard.

File: src/main/java/com/owncloud/android/media/MediaControlView.java [hunk 2]
```diff
@@ -223,10 +223,10 @@ private void initControllerView(View v) {
         if (mProgress != null) {
             if (mProgress instanceof SeekBar) {
                 SeekBar seeker = (SeekBar) mProgress;
-                ThemeUtils.colorPreLollipopHorizontalSeekBar(seeker);
+                ThemeUtils.colorHorizontalSeekBar(seeker);
                 seeker.setOnSeekBarChangeListener(this);
             } else {
-                ThemeUtils.colorPreLollipopHorizontalProgressBar(mProgress);
+                ThemeUtils.colorHorizontalProgressBar(mProgress, ThemeUtils.primaryAccentColor());
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/media/MediaControlView.java]: This has nothing to do with the scope of the PR, so this should idealistically be in a new PR/branch to allow faster and easier reviews.

File: src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.java [hunk 3]
```diff
@@ -514,6 +531,88 @@ private void setTimeModified(long milliseconds) {
...
+            noList.setVisibility(View.GONE);
+            setListViewHeightBasedOnChildren(usersList);
+
+        } else {
+            usersList.setVisibility(View.GONE);
+            noList.setVisibility(View.VISIBLE);
+        }
+    }
+
+    /* Fix scroll in listview when the parent is ac ScrollView */
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.java]: typo "ac"

File: src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.java [hunk 4]
```diff
@@ -514,6 +531,88 @@ private void setTimeModified(long milliseconds) {
...
+                view.setLayoutParams(new ViewGroup.LayoutParams(desiredWidth, ViewGroup.LayoutParams.WRAP_CONTENT));
+            }
+            view.measure(desiredWidth, View.MeasureSpec.UNSPECIFIED);
+            totalHeight += view.getMeasuredHeight();
+        }
+        ViewGroup.LayoutParams params = listView.getLayoutParams();
+        params.height = totalHeight + (listView.getDividerHeight() * (listAdapter.getCount() - 1));
+        listView.setLayoutParams(params);
+        listView.requestLayout();
+    }
```
Comment (tobiasKaminsky) [inline src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.java]: Isn't this possible with proper xml settings? ListView should "just" take all remaining space?

**Comment by tobiasKaminsky** [comment]
:+1: 
only some minor remarks

File: src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.java (hunk 4 above)
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/ui/fragment/FileDetailFragment.java]: Unfortunately no. Filling the remaining screen space isn't enough. The List view needs to have the size of the content and wrap_content also only makes the list stretch to the screen size limit which the adds the problem with to scroll-able views... So dynamically calculate the size is actually the only option that seems to be working properly right now.

File: src/main/java/com/owncloud/android/media/MediaControlView.java (hunk 2 above)
Comment (AndyScherzinger) [inline src/main/java/com/owncloud/android/media/MediaControlView.java]: So you want me to cut it out of this PR?

**Comment by AndyScherzinger** [comment]
@mario are you fine with the content of this PR?

**Comment by AndyScherzinger** [comment]
@mario @tobiasKaminsky are you fine with the PR? (did another rebase and some further lint fixes)

**Comment by mario** [comment]
Conflict, but otherwise nice.

**Comment by AndyScherzinger** [comment]
@mario conflict will be solved and PR can be merged after the master has been build by drone again, this branch has been rebased again and drone has successfully run again 🎉  Lint is really painfully build/merge wise...

**Comment by AndyScherzinger** [comment]
@mario @tobiasKaminsky build turned green, can now be merged. Fire at will :)

**Comment by mario** [comment]
Do it

**Pull request merged**